src/frontend/parallelization_phase.nit

```diff
@@ -65,6 +60,15 @@ private class ParallelizationPhase
 			classname += nmethdef.n_methid.as(AIdMethid).n_id.text
 		end
 
+		# Handle methods with a return value
+		var has_rvalue = nmethdef.n_signature.n_type != null
+		var vtype = ""
+		var rstatement = ""
+		if has_rvalue then
+			vtype = "redef type E: " + nmethdef.n_signature.n_type.n_id.text
+			rstatement = "return thread"
```

Comment:
As said in the previous comment, I think that you shoud return a thread object even on precedure so that user can check for is_done and/or join